synedrion/src/uint/signed.rs

```diff
@@ -287,42 +364,68 @@ impl<T: UintLike + HasWide> Signed<T> {
         bound_bits: usize,
         scale: &Bounded<T>,
     ) -> Signed<T::Wide> {
-        assert!(bound_bits < <T as Integer>::BITS - 1);
-        let scaled_bound = scale.as_ref().into_wide().shl_vartime(bound_bits);
+        let bound_bits = u32::try_from(bound_bits).expect("Assumed to fit in a u32; caller beware");
+        assert!(
+            bound_bits < T::BITS - 1,
+            "Out of bounds: bound_bits was {} but must be smaller than {}",
+            bound_bits,
+            T::BITS - 1
+        );
+        let scaled_bound = scale
+            .as_ref()
+            .clone()
+            .into_wide()
+            .overflowing_shl_vartime(bound_bits)
+            .expect("Just asserted that bound bits is smaller than T's bit precision");
 
         // Sampling in range [0, 2^bound_bits * scale * 2 + 1) and translating to the desired range.
         let positive_bound = scaled_bound
-            .shl_vartime(1)
-            .checked_add(&T::Wide::ONE)
-            .unwrap();
-        let positive_result = T::Wide::random_mod(rng, &NonZero::new(positive_bound).unwrap());
+            .overflowing_shl_vartime(1)
+            .expect("TODO: justify this properly")
```

Comment:
Probably better leave as `unwrap()` and add a `TODO (#76)` - will be more noticeable when we start fixing that issue.

Reply:
This is a "note to self" kind of `TODO`, one I intend to fix in this PR before merging; I missed this one before asking for a review, apologies!

I think `expect("why this will never (ever) happen");` is a good pattern generally, because it really forces the author to pause and reflect on why it is ok to unwrap.